src/runtime/pprof/pprof.go

```diff
@@ -776,6 +779,107 @@ func writeGoroutineStacks(w io.Writer) error {
 	return err
 }
 
+// writeGoroutineStacksWithLabels writes individual goroutine stack traces like
+// writeGoroutineStacks, but uses the concurrent, reduced-stop time approach of
+// goroutineProfileWithLabelsConcurrent. It includes the ID, state, and labels
+// for each goroutine, although as these are captured after the world is resumed
+// they are not guaranteed to be entirely consistent as of a single point in
+// time.
+func writeGoroutineStacksWithLabels(w io.Writer) error {
+	n, ok := pprof_goroutineStacksWithLabels(nil, nil, nil)
+
+	for {
+		// Allocate slices for individual goroutine data
+		stacks := make([]profilerecord.StackRecord, n+10)
+		labels := make([]unsafe.Pointer, n+10)
+		infos := make([]profilerecord.GoroutineInfo, n+10)
+
+		n, ok = pprof_goroutineStacksWithLabels(stacks, labels, infos)
+		if ok {
+			return printGoroutineStacksWithLabels(w, stacks[:n], labels[:n], infos[:n])
+		}
+		// Profile grew; try again with larger slices
+	}
+}
+
+// printGoroutineStacksWithLabels formats goroutine records in a format similar
+// to runtime.Stack, but also includes labels as well.
+func printGoroutineStacksWithLabels(
```

Comment:
Is there a smaller diff that takes the existing print code and adds an argument for it?